### PR TITLE
lightning: remove error check in `TestAdjustBlockSize`

### LINKED ISSUE #52277: require no error in `TestAdjustBlockSize`
## Enhancement
The `dial tcp 127.0.0.1:10080: connect: connection refuse` error checked in test `TestAdjustBlockSize` has been tested by `TestAdjustConnectRefused`. We can remove this check.

### PR DESCRIPTION
**Pull request by lyzx2001**
### What problem does this PR solve?

Issue Number: close #52277 

Problem Summary:
The `dial tcp 127.0.0.1:10080: connect: connection refuse` error checked in test `TestAdjustBlockSize` has been tested by `TestAdjustConnectRefused`.

### What changed and how does it work?
The error check has been removed.
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/lightning/config/config_test.go [hunk 1]
```diff
@@ -1322,16 +1322,21 @@ func TestAdjustConflict(t *testing.T) {
 }
 
 func TestAdjustBlockSize(t *testing.T) {
+	ts, host, port := startMockServer(t, http.StatusOK,
+		`{"port":6666,"advertise-address":"121.212.121.212:5555","path":"34.34.34.34:3434"}`,
```
[the remark this answers is not in the document]
Comment (lyzx2001) [reply br/pkg/lightning/config/config_test.go:1326]: 0 port is not supported here. The expected error of 0 port is "[Lightning:Config:ErrInvalidConfig]invalid `tidb.port` setting"

[the remark this answers is not in the document]
Comment (lance6716) [reply br/pkg/lightning/config/config_test.go:1326]: > Can you use 0 port to random the port?

the port in mock content is not accessed by lightning.

**Review by lance6716 — APPROVED** [review]

File: pkg/lightning/config/config_test.go [hunk 2]
```diff
@@ -1322,16 +1322,21 @@ func TestAdjustConflict(t *testing.T) {
...
 	cfg.TiDB.DistSQLScanConcurrency = 1
 	cfg.Mydumper.SourceDir = "."
 	cfg.TikvImporter.BlockSize = 0
 
 	err := cfg.Adjust(context.Background())
-	// should return "dial tcp 127.0.0.1:10080: connect: connection refused" error,
-	// if you run it locally, shutdown the tidb server first.
-	require.Error(t, err)
+	require.NoError(t, err)
 	require.Equal(t, ByteSize(16384), cfg.TikvImporter.BlockSize)
```
[the remark this answers is not in the document]
Comment (lyzx2001) [reply pkg/lightning/config/config_test.go]: modified

**Review by D3Hunter — APPROVED** [review]

**Pull request merged**